tests/cupy_tests/indexing_tests/test_indexing.py

```diff
@@ -29,6 +31,15 @@ def test_take_no_axis(self, xp):
         b = xp.array([[10, 5], [3, 20]])
         return a.take(b)
 
+    # see cupy#3017
+    @testing.for_dtypes('bBhH')
+    @testing.numpy_cupy_array_equal()
+    def test_take_index_range(self, xp, dtype):
+        iinfo = numpy.iinfo(dtype)
+        a = testing.shaped_arange((iinfo.max + 1,), xp)
+        b = testing.shaped_random((1,), xp, dtype=dtype, scale=iinfo.max)
```

Comment:
Let's use a concrete value instead of `shaped_random`.

Reply:
Fixed!